qlib/SftpPoller.qm

```diff
@@ -499,6 +515,37 @@ public namespace SftpPoller {
             return sftp.stat(path, n_timeout ?? timeout);
         }
 
+        #! check if a remote path is writable
+        /**
+            @throw "REMOTE-DIR-ERROR" this exception is thrown if the remote path does not exist, is not a directory, or is not not writable
+        */
+        checkRemotePath(string path, bool write = False, *timeout n_timeout) {
+            hash h = stat(path, n_timeout);
+            string apath = rootSftpPath + DirSep + path;
+            if (!h)
+                throw "REMOTE-DIR-ERROR", sprintf("%y: path does not exist", apath);
+            string perms = h.permissions;
+            if (perms[0] != "d")
+                throw "REMOTE-DIR-ERROR", sprintf("%y: path is not a directory", apath);
+
+            string f = path + DirSep + check_file;
+            if (stat(f, n_timeout))
+                sftp.removeFile(f, n_timeout);
```

Comment:
if the file exists, then exit immediately after deleting - deleting the file is proof that the directory is writable; there is no need to write the file and then delete it again